Panic in AWS Node Attestor

I saw this panic last night but haven't investigated further.  Filing an issue to keep track

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb175d8]

goroutine 39 [running]:
github.com/aws/aws-sdk-go/aws/request.(*Request).ApplyOptions(0xc00015d000, 0xc00012e120, 0x1, 0x1)
	/go/src/github.com/spiffe/spire/vendor/github.com/aws/aws-sdk-go/aws/request/request.go:192 +0x38
github.com/aws/aws-sdk-go/service/ec2.(*EC2).DescribeInstancesWithContext(0xc00012e110, 0x7f85461aa168, 0xc00012a500, 0xc00013dd10, 0xc00012e120, 0x1, 0x1, 0x0, 0x0, 0x1)
	/go/src/github.com/spiffe/spire/vendor/github.com/aws/aws-sdk-go/service/ec2/api.go:10995 +0x18b
github.com/spiffe/spire/pkg/server/plugin/nodeattestor/aws.(*IIDAttestorPlugin).ec2Attestation(0xc0003ba840, 0x1fdfb60, 0xc00012a500, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/spiffe/spire/pkg/server/plugin/nodeattestor/aws/iid.go:253 +0x38c
github.com/spiffe/spire/pkg/server/plugin/nodeattestor/aws.(*IIDAttestorPlugin).Attest(0xc0003ba840, 0x1fd86a0, 0xc0001182c0, 0xc0001182c0, 0x1)
	/go/src/github.com/spiffe/spire/pkg/server/plugin/nodeattestor/aws/iid.go:139 +0xb7b
github.com/spiffe/spire/proto/server/nodeattestor.BuiltIn.Attest.func1(0x1fe01a0, 0xc00038c690, 0x1fcc160, 0xc0003ba840)
	/go/src/github.com/spiffe/spire/proto/server/nodeattestor/nodeattestor.go:98 +0x89
created by github.com/spiffe/spire/proto/server/nodeattestor.BuiltIn.Attest
	/go/src/github.com/spiffe/spire/proto/server/nodeattestor/nodeattestor.go:97 +0x1a7
```

--- a/pkg/server/plugin/nodeattestor/aws/iid.go
+++ b/pkg/server/plugin/nodeattestor/aws/iid.go
@@ -250,7 +250,7 @@ func (p *IIDAttestorPlugin) ec2Attestation(ctx context.Context, c IIDAttestorCon
 		InstanceIds: []*string{&doc.InstanceID},
 	}
 
-	result, err := ec2Client.DescribeInstancesWithContext(ctx, query, nil)
+	result, err := ec2Client.DescribeInstancesWithContext(ctx, query)
 	if err != nil {
 		return caws.AttestationStepError("querying AWS via describe-instances", err)
 	}

--- a/pkg/server/plugin/nodeattestor/aws/iid_test.go
+++ b/pkg/server/plugin/nodeattestor/aws/iid_test.go
@@ -188,7 +188,7 @@ func (s *IIDAttestorSuite) TestClientAndIDReturns() {
 			mockExpect: func(mock *mock_aws.MockEC2Client) {
 				mock.EXPECT().DescribeInstancesWithContext(gomock.Any(), &ec2.DescribeInstancesInput{
 					InstanceIds: []*string{&testInstance},
-				}, nil).Return(nil, errors.New("client error"))
+				}).Return(nil, errors.New("client error"))
 			},
 			expectErr: "client error",
 		},
@@ -200,7 +200,7 @@ func (s *IIDAttestorSuite) TestClientAndIDReturns() {
 				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &nonzeroDeviceIndex
 				mock.EXPECT().DescribeInstancesWithContext(gomock.Any(), &ec2.DescribeInstancesInput{
 					InstanceIds: []*string{&testInstance},
-				}, nil).Return(&output, nil)
+				}).Return(&output, nil)
 			},
 			expectErr: "verifying the EC2 instance's NetworkInterface[0].DeviceIndex is 0",
 		},
@@ -212,7 +212,7 @@ func (s *IIDAttestorSuite) TestClientAndIDReturns() {
 				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
 				mock.EXPECT().DescribeInstancesWithContext(gomock.Any(), &ec2.DescribeInstancesInput{
 					InstanceIds: []*string{&testInstance},
-				}, nil).Return(&output, nil)
+				}).Return(&output, nil)
 			},
 			skipBlockDev: true,
 			expectID:     "spiffe://example.org/spire/agent/aws_iid/test-account/test-region/test-instance",
@@ -230,7 +230,7 @@ func (s *IIDAttestorSuite) TestClientAndIDReturns() {
 				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
 				mock.EXPECT().DescribeInstancesWithContext(gomock.Any(), &ec2.DescribeInstancesInput{
 					InstanceIds: []*string{&testInstance},
-				}, nil).Return(&output, nil)
+				}).Return(&output, nil)
 			},
 			expectID: "spiffe://example.org/spire/agent/aws_iid/test-account/test-region/test-instance",
 		},
@@ -242,7 +242,7 @@ func (s *IIDAttestorSuite) TestClientAndIDReturns() {
 				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
 				mock.EXPECT().DescribeInstancesWithContext(gomock.Any(), &ec2.DescribeInstancesInput{
 					InstanceIds: []*string{&testInstance},
-				}, nil).Return(&output, nil)
+				}).Return(&output, nil)
 			},
 			replacementTemplate: "{{ .PluginName}}/{{ .Region }}/{{ .AccountID }}/{{ .InstanceID }}",
 			expectID:            "spiffe://example.org/spire/agent/aws_iid/test-region/test-account/test-instance",

